src/thor/unidirectional_astar.cc

```diff
@@ -603,6 +611,9 @@ template <const ExpansionType expansion_direction, const bool FORWARD>
 void UnidirectionalAStar<expansion_direction, FORWARD>::ModifyHierarchyLimits(
     const float dist,
     const uint32_t /*density*/) {
+
+  if (!costing_->DefaultHierarchyLimits())
```

Comment:
If the user passed in any custom hierarchy limits, don't do this distance based modification.